pwa/src/components/endpoints/form.tsx

```diff
@@ -0,0 +1,241 @@
+      name: event.target.name.value,
+      description: event.target.description.value ? event.target.description.value : null,
+      type: event.target.type.value,
+      path: event.target.path.value,
+      application: event.target.application.value ? event.target.application.value : null,
+    };
+
+    body = removeEmptyObjectValues(body);
+    if (!checkValues([body.name, body.type, body.path])) {
+      setAlert(null);
+      setAlert({ type: 'danger', message: 'Required fields are empty' });
+      setShowSpinner(false);
+      return;
+    }
+
+    let url = `${context.adminUrl}/endpoints`;
+    let method = "POST";
+    if (id !== "new") {
+      url = `${url}/${id}`;
+      method = "PUT";
+    }
+
+    fetch(url, {
+      method: method,
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: "Bearer " + sessionStorage.getItem("jwt"),
+      },
+      body: JSON.stringify(body),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        // console.log("post endpoint")
+        // console.log(data)
+        setShowSpinner(false);
+        setEndpoint(data)
+        method === 'POST' && navigate("/endpoints")
+      })
+      .catch((error) => {
+        setShowSpinner(false);
+        console.error("Error:", error);
+        setAlert(null);
+        setAlert({ type: 'danger', message: error.message });
+      });
+  };
+
+  return (<>
+      {
+        alert !== null &&
+        <FlashMessage duration={5000}>
+          <Alert alertClass={alert.type} body={function () { return (<>{alert.message}</>) }} />
+        </FlashMessage>
+      }
+      <form id="dataForm" onSubmit={saveEndpoint}>
+        <Card title="Values"
+              cardHeader={function () {
+                return (<>
+                  <Link className="utrecht-link" to={"/endpoints"}>
+                    <button className="utrecht-button utrecht-button-sm btn-sm btn-danger mr-2">
+                      <i className="fas fa-long-arrow-alt-left mr-2" />Back
+                    </button>
+                  </Link>
+                  <button
+                    className="utrecht-button utrec`ht-button-sm btn-sm btn-success"
+                    type="submit"
+                  >
+                    <i className="fas fa-save mr-2" />Save
+                  </button>
+                </>)
+              }}
+              cardBody={function () {
+                return (
+                  <div className="row">
+                    <div className="col-12">
+                      {showSpinner === true ? (
+                        <Spinner />
+                      ) : (
+                        <>
+                          <div className="row">
+                            <div className="col-6">
+                              {endpoint !== null && endpoint.name !== null ? (
+                                <GenericInputComponent type={"text"} name={"name"} id={"nameInput"} data={endpoint.name}
```

Comment:
Ik denk dat we even goed moeten nadenken over dit `GenericInputComponent`, misschien dat we beter af zijn zonder zo'n complexe implementatie van uiteindelijk simpele velden.